File: Data Structures & Algorithms/duplicate-integer/submission-0.kt
class Solution {
    fun hasDuplicate(nums: IntArray): Boolean {
        var result = false
        for(i in 0 until nums.size) {
            var case = nums[i]
            for (n in i + 1 until nums.size) {
                if (case == nums[n]) result = true
            }
        }
        return result
    }
}

File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> ranking;
        for(int elem : nums) {
            auto it = ranking.find(elem);
            if (it != ranking.end()) {
                it->second += 1;
            } else {
                ranking.insert({elem, 1});
            }
        }
        vector<int> result;
        int biggest = 0;
        int saveKey = 0;
        while (k > 0) {
            biggest = 0;
            for (auto& [key, value] : ranking) {
                if (value > biggest) {
                    biggest = value;
                    saveKey = key;
                }
            }
            result.push_back(saveKey);
            ranking.erase(saveKey);
            k--;
        }
        return result;
    }
};
